dcpy/models/product/dataset/metadata.py

```diff
@@ -177,15 +177,29 @@ class Metadata(BaseModel, extra="forbid"):
     package: Package
     columns: list[Column]
 
+    # Hold onto this for serialization, to avoid Pydantics reformatting of the metadata.yml
+    _templated_source_metadata: str
```

Comment:
Just curious because I've thoughts on this too - what mainly here? None values in the output, reordering, loss of comments, etc

Reply:
At some point it would be worth making our own subclass of BaseClass to have a little more fine-grained control over some of this, and have things like having all datetimes serialize the way we want, enums serializing to values by default, etc

Reply:
Yeah, it's all of those things, esp reordering. I'm using this to get the templated metadata into the package, but that said, my broader concern here is actually whether we _should_ be doing that at all. Or does it just introduce potential new problems, e.g. what if we change a Socrata destination four-four... do we really want stale product metadata hanging around. I'm unsure.

Reply:
I think it makes sense that a specific packaged or distributed version of a dataset has version-specific metadata. Ideally it doesn't become stale but it seems unavoidable.

That being said, it also doesn't seem super wrong to have the untemplated metadata. Depends what's being templated I suppose. There's an argument `version` field (or some field in general that is focused on any metadata specific to a given version/build, or a separate file with this information that gets templated) and then leave fields untemplated to keep the mutable and immutable fields a little more separate. I don't know 🤷 

If we keep templating, I think this also ties back into a custom class - would be nice to allow fairly simple validation around what fields can be templated or not.

Reply:
writing this down for my records. I'm thinking that manually preserving yaml formtting is a losing game, and we can do most of what we (or just I) need by 
- We can order top-level keys in the yaml like this `yaml.dump(data, default_flow_style=False, sort_keys=False)`
- I like having a newline between top-sections (makes it easier for me to traverse) which we can do with something like [this](https://github.com/yaml/pyyaml/issues/127#issuecomment-525800484)